owrrm/src/main/java/com/facebook/openwifi/rrm/optimizers/clientsteering/ClientSteeringOptimizer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifi.rrm.optimizers.clientsteering;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import com.facebook.openwifi.rrm.DeviceConfig;
+import com.facebook.openwifi.rrm.DeviceDataManager;
+import com.facebook.openwifi.rrm.modules.Modeler.DataModel;
+
+/** Client steering base class */
+public abstract class ClientSteeringOptimizer {
+	// TODO call upon triggers, not only via one-off or period runs
+
+	/** Represents client steering actions an AP Can take */
+	public static enum CLIENT_STEERING_ACTIONS {
+		/** Steer from 2G to 5G/6G */
+		STEER_UP,
+		/** Steer from 5G/6G to 2G */
+		STEER_DOWN,
+		/** Deauthenticate client */
+		DEAUTHENTICATE
+	}
+
+	/**
+	 * Map from AP serial number to radio bssid to time (unix time in ms) of the
+	 * latest attempted client steering action. It is static here to track
+	 * attempts across all instances of all subclasses. The {@code Long} values
+	 * are never null.
+	 */
+	private static final Map<String, Map<String, Long>> apRadioLastAttempt =
+		new TreeMap<>();
+
+	/**
+	 * Register the time of the latest client steering attempt by AP serial
+	 * number and radio bssid.
+	 */
+	protected static final void registerClientSteeringAttempt(
+		String apSerialNumber,
+		String bssid
+	) {
+		apRadioLastAttempt
+			.computeIfAbsent(apSerialNumber, k -> new TreeMap<>())
+			.put(bssid, System.currentTimeMillis());
+	}
+
+	/**
+	 * Get the time of the latest client steering attempt by AP serial number
+	 * and radio bssid. Return {@link Long#MIN_VALUE} if no client steering
+	 * attempt has been made for the given AP and radio.
+	 */
+	protected static final long getLatestClientSteeringAttempt(
```

Comment:
just return boxed `Long` so you can use null? don't use `Long.MIN_VALUE`, that doesn't make sense